graphql_compiler/compiler/emit_sql.py

```diff
@@ -224,43 +242,72 @@ def _construct_fold_subquery(self, subquery_from_clause):
+        # keep track of whether or not an _x_count has been added to the outputs
+        has_x_count = False
+
         # find outputs for this fold in all_folded_outputs and add to self._outputs
         if fold_scope_location.fold_path in all_folded_outputs:
             for fold_output in all_folded_outputs[fold_scope_location.fold_path]:
                 # distinguish between folds with the same fold path but different query paths
                 if (fold_output.base_location, fold_output.fold_path) == (
                         fold_scope_location.base_location, fold_scope_location.fold_path):
-                    if fold_output.field == '_x_count':
-                        raise NotImplementedError(u'_x_count not implemented in SQL')
-
-                    # force column to have explicit label as opposed to anon_label
-                    intermediate_fold_output_name = 'fold_output_' + fold_output.field
-                    # add array aggregated output column to self._outputs
-                    self._outputs.append(
-                        sqlalchemy.func.array_agg(
-                            self.output_vertex_alias.c[fold_output.field]
-                        ).label(intermediate_fold_output_name)
-                    )
+                    if fold_output.field == COUNT_META_FIELD_NAME:
+                        has_x_count = True
+                        self._outputs.append(
+                            sqlalchemy.func.coalesce(
+                                sqlalchemy.func.count(),
+                                sqlalchemy.literal_column('0')
+                            ).label('fold_output_x_count')
+                        )
+
+                    else:
+                        # force column to have explicit label as opposed to anon_label
+                        intermediate_fold_output_name = 'fold_output_' + fold_output.field
+                        # add array aggregated output column to self._outputs
+                        self._outputs.append(
+                            sqlalchemy.func.array_agg(
+                                self.output_vertex_alias.c[fold_output.field]
+                            ).label(intermediate_fold_output_name)
+                        )
 
         # use to join unique identifier for the fold's outer vertex to the final table
         self._outputs.append(self.outer_vertex_alias.c[join_descriptor.from_column])
 
+        # _x_count must be added if _x_count is used in filtering, but is not an output
+        if not has_x_count:
+            for x_count_filter in x_count_filters:
+                if (x_count_filter.fold_scope_location.at_vertex() ==
+                        fold_scope_location.at_vertex()):
+                    self._outputs.append(
+                        sqlalchemy.func.coalesce(
+                            sqlalchemy.func.count(),
+                            sqlalchemy.literal_column('0')
+                        ).label('fold_output_x_count')
```

Comment:
Does this mean `fold_output_x_count` is disallowed as an `out_name` of a folded output?

Reply:
This would actually be a problem with `x_count` since `fold_output_` is prepended to the `out_name` to get the intermediate fold output name. I will change this label to `fold_output__x_count` so that it follows the convention of the others.

Reply:
Aah I see. Can you explan this in a comment next to the definition of the `fold_output__x_count` constant?

Reply:
Switched to constructing this string with constants. Hopefully it is a little clearer now!